TrenchBroomTest/raylib_viewer.odin

```
@@ -0,0 +1,471 @@
+
+		// Create a model from the mesh and apply texture if available
+		model := rl.LoadModelFromMesh(raylib_mesh)
+
+		// Extract material name from the quake mesh
+		material_name := extract_material_name_from_mesh(quake_mesh)
+		append(&mesh_texture_names, material_name)
+
+		// Try to apply the texture
+		if texture, has_texture := loaded_textures[material_name]; has_texture {
+			model.materials[0].maps[rl.MaterialMapIndex.ALBEDO].texture = texture
+			fmt.printf("Applied texture '%s' to world mesh\n", material_name)
+		} else {
+			model.materials[0].maps[rl.MaterialMapIndex.ALBEDO].color = rl.WHITE
+			fmt.printf("No texture found for material '%s', using white color\n", material_name)
+		}
+
+		append(&raylib_models, model)
+	}
+
+	// Convert entity geometry
+	for quake_mesh in loaded_map.entity_geometry {
+		if len(quake_mesh.vertices) == 0 || len(quake_mesh.indices) == 0 {
+			continue
+		}
+
+		raylib_mesh := convert_mesh_to_raylib(quake_mesh)
+		append(&raylib_meshes, raylib_mesh)
+
+		// Create a model from the mesh and apply texture if available
+		model := rl.LoadModelFromMesh(raylib_mesh)
+
+		// Extract material name from the quake mesh
+		material_name := extract_material_name_from_mesh(quake_mesh)
+		append(&mesh_texture_names, material_name)
+
+		// Try to apply the texture (entities use same logic as world)
+		if texture, has_texture := loaded_textures[material_name]; has_texture {
+			model.materials[0].maps[rl.MaterialMapIndex.ALBEDO].texture = texture
+			fmt.printf("Applied texture '%s' to entity mesh\n", material_name)
+		} else {
+			model.materials[0].maps[rl.MaterialMapIndex.ALBEDO].color = rl.GREEN
+			fmt.printf(
+				"No texture found for material '%s', using green color for entity\n",
+				material_name,
+			)
+		}
+
+		append(&raylib_models, model)
+	}
+
+	fmt.printf("Converted %d meshes to Raylib\n", len(raylib_meshes))
+}
+
+convert_mesh_to_raylib :: proc(quake_mesh: quakemap.Mesh) -> rl.Mesh {
```

Whats difference between this proc and  `convert_quake_meshes_to_raylib`? Names are similar

maybe this should be in the quakemap lib too?

---

This should return an error enum